python/lib/dependabot/python/update_checker/pipenv_version_resolver.rb

```diff
@@ -232,10 +234,36 @@ def handle_pipenv_errors_resolving_original_reqs(error)
             raise DependencyFileNotResolvable, msg
           end
 
+          # NOTE: Pipenv masks the actualy error, see this issue for updates:
+          # https://github.com/pypa/pipenv/issues/2791
+          pipenv_installation_error = "pipenv.patched.notpip._internal."\
+                                      "exceptions.InstallationError: "\
+                                      "Command \"python setup.py egg_info\" "\
+                                      "failed with error code 1 in"
+          if error.message.include?(pipenv_installation_error)
+            # Find the dependency that's causing resolution to fail
+            escaped_pipenv_error = Regexp.quote(pipenv_installation_error)
+            name_regex = %r{#{escaped_pipenv_error}.+/(?<name>.+)/$}.freeze
+            dependency_name = error.message.match(name_regex).
+                              named_captures["name"]
+            raise unless dependency_name
+
+            msg = "Pipenv failed to install \"#{dependency_name}\". "\
+                  "This could be caused by missing system "\
+                  "dependencies that can't be installed by Dependabot or "\
+                  "required install flags.\n\n"\
+                  "Error output from running \"pipenv lock\":\n"\
+                  "#{clean_error_message(error.message)}"
```

Comment:
Could we pull some of this out into a method instead of disabling `AbcSize` and `MethodLength`?